### ISSUE TITLE
Slight modification to FIRST/LASTNIGHT definitions to be "file on disk" oriented

### ISSUE THREAD

**Issue by stephjuneau** [desihub/desispec#2069]
After comparing the content of EXP_FIBERMAP extensions from various catalogs to the summary `zall`  files and namely the zall-tilecumulative version, I realized that `LASTNIGHT` is already used to signify the last night during which data were obtained (regardless of the data quality and therefore regardless of whether the exposures will be included in a coadd). Therefore, I propose the following definitions for FIRSTNIGHT, LASTNIGHT to ensure that LASTNIGHT remains identical and FIRSTNIGHT is its logical equivalent:

```
FIRSTNIGHT (int32) =  KPNO calendar date when the first exposures was obtained (regardless of data quality)
LASTNIGHT (int32) =  KPNO calendar date when the last exposures was obtained (regardless of data quality)
```

Relative to PR https://github.com/desihub/desispec/pull/2067, this will leave the `MJD` columns unchanged because those correspond to the photons/exposures that were used to generate the coadd ("good_coadds") consistently with MEAN_FIBER_RA, DEC. etc.

In practice, the NIGHT and MJD columns will closely track one another most of the time. But there will be exceptions if the first or last night of data happen to be excluded from the coadds and also due to the NIGHT being the KPNO definition of calendar date that rolls over at noon rather than midnight.

To be extra clear:
```
MIN_MJD (float64) = Minimum value of the Modified Julian Date (when the shutter was open for the first exposure used in the coadded spectrum)
MEAN_MJD (float64) = Mean value across exposures of the Modified Julian Date (when the shutter was open for each exposure used in the coadded spectrum)
MAX_MJD (float64) = Maximum value of the Modified Julian Date (when the shutter was open for the last exposure used in the coadded spectrum)
```

In other words, `MJD` columns are photon-based in the sense of capturing when the coadded photons were collected, whereas the `NIGHT` columns are "files on disk"-based (regardless of data quality). If folks agree, I will create a new branch and make that change soon to keep the zcat VACs consistent with the pre-existing definition of LASTNIGHT. (Cc @sbailey @weaverba137 @moustakas @araichoor @geordie666)

**Comment by moustakas** [comment]
This looks very sensible.

**Comment by sbailey** [comment]
@stephjuneau good catch, I agree with your assessment.  Please proceed with a PR with the change.
